### ISSUE TITLE
Investigate possibility of moving custom JSAPI build to a bower build

### ISSUE THREAD

**Issue by nheminger** [Esri/ago-assistant#113]
Esri Javascript API (JSAPI) 3.15 is [released](http://blogs.esri.com/esri/arcgis/2015/12/01/arcgis-api-for-javascript-version-3-15-released/) and now includes [bower custom builds](https://developers.arcgis.com/javascript/jshelp/inside_bower_custom_builds.html). Additional [Esri JSAPI bower tools](https://github.com/Esri/jsapi-resources/tree/master/bower) may be useful. It might be useful to investigate moving the custom JSAPI build as a bower build step in the grunt build file?

**Comment by ecaldwell** [comment]
Yes, great idea. I'd actually like to make this part of a much larger push to truly "build" the app. As it stands, the build is only modestly smaller and still includes separate files for each library. There's a lot of opportunity for further concatenation.

Also, would that impact any of the work you did in #102 or would all of those files just stay put?

**Comment by nheminger** [comment]
Definitely, that makes sense! I noticed there are a few 404 errors with missing files too, which may be an issue with the build. I think they're mostly icons that aren't essential to functionality, but a larger issue to look at a truly "built" version may fix those errors and just make everything more compact and slick for development :sunglasses:.

Yep, I definitely think it could affect the work I did for #102 so this would need to take that into account and also re-test locales. Even if the JSAPI stays with the online optimizer build, the work I did there could be possibly optimized more.

I don't know if the app's "built" version can/should include all locales as there may be a lot of locales and it may be better how the app conditionally downloads "other" locales like it does now. It may be worth checking though if they can be be included by default if their aggregation is small enough.

I think any revised bower or other build will need to specifically take into account locales and how those are going to be built correctly to work.

**Comment by slibby** [comment]
No longer relevant
